### PR TITLE
Add slda

### PR DESCRIPTION
**Pull request by mmb90**

### CODE REVIEW

File: recipes/slda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+build:
+    number: 0
+    skip: True  # [win or not py3k]
+
+requirements:
+    build:
+        - cython
+        - cythongsl
+        - gcc
```
Comment (ocefpaf) [inline recipes/slda/meta.yaml]: Do you need Fortran or `libgmop`? If not you probably don't need `gcc` here and `libgcc` below.

**Comment by ocefpaf** [comment]
@mmb90 the failure you are seeing has nothing to do with your PR and a fix is under way.

File: recipes/slda/meta.yaml (hunk 1 above)
Comment (mmb90) [inline recipes/slda/meta.yaml]: You're right I don't need them. They are fossils from when I had to build a supporting package at the same time (which is now in conda-forge).

File: recipes/slda/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        - cython
+        - cythongsl
+        - gsl
+        - numpy x.x
+        - pypolyagamma
+        - python
+        - scikit-learn
+        - scipy
+        - setuptools
+
```
Comment (ocefpaf) [inline recipes/slda/meta.yaml]: This is a minor thing, but can you remove the extra blank line here?

File: recipes/slda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    run:
+        - cython
+        - cythongsl
+        - gsl
+        - numpy x.x
+        - pypolyagamma
+        - python
+        - scikit-learn
+        - scipy
+        - setuptools
```
Comment (ocefpaf) [inline recipes/slda/meta.yaml]: Do you know if `setuptools` is needed at run-time? (Maybe something import `pkg_resources`?) If not can you removed it?

Comment (mmb90) [inline recipes/slda/meta.yaml]: It uses pkg_resources.

File: recipes/slda/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+build:
+    number: 0
+    skip: True  # [win or not py3k]
+
+requirements:
+    build:
+        - cython
+        - cythongsl
+        - gsl
```
Comment (ocefpaf) [inline recipes/slda/meta.yaml]: `cythongsl` depends on `gsl`. To avoid over-specifying the dependencies you can remove `gsl` from both `build` and `run` sections.

File: recipes/slda/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/slda/meta.yaml]: So, this is really a consequence of not having proper [`entrypoints`]( http://conda.pydata.org/docs/building/meta-yaml.html#python-entry-points ). If we add those, we should not need `setuptools` at run-time.

Comment (ocefpaf) [inline recipes/slda/meta.yaml]: > It uses pkg_resources.

Cool. Just leave it then. Thanks for the clarification!

File: recipes/slda/build.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+export USE_CYTHON=True
+
+export CC=gcc CXX=g++
```
Comment (jakirkham) [inline recipes/slda/build.sh]: So, this is going to cause issues if we do this for both platforms. I expect either this isn't needed or we need to use something like the `toolchain`.

Comment (mmb90) [inline recipes/slda/build.sh]: Yeah, we shouldn't need to do that export anymore. It's a fossil from when I had to build a supporting package at the same time (which is now in conda-forge).

**Comment by mmb90** [comment]
Forgot to update the release in the soda repository. Will fix that.

**Comment by mmb90** [comment]
> the failure you are seeing has nothing to do with your PR and a fix is under way.

@ocefpaf, are there updates on this?

**Comment by ocefpaf** [comment]
Can you rebase your PR against the latest master? That should fix it.

**Comment by patricksnape** [comment]
Looks like `cythongsl` doesn't exist for [Windows Python 2.x](https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.4432/job/536mc5wu27v3e7hb#L198)

**Comment by ocefpaf** [comment]
> Looks like Appveyor was still triggered despite the `skip`

That is a bug when we mix latest `conda-build` and a previous versions of `conda-build-all`. The pre-processor is not there b/c `conda-build` did not render the recipe when `conda-build-all` expected.

That is fixed, but another bug is preventing us from releasing `conda-build-all` on all Archs+Pythons on Windows. (Sorry, I forgot the xrefs.)

TL;DR rebasing will get rid of that recipe and the rest is good to go.

**Comment by mmb90** [comment]
I've tried rebasing onto the latest master and pushing to the PR again, still no luck.

**Comment by jakirkham** [comment]
The `conda-build-all` deployment issue has been resolved. Restarting AppVeyor.

File: recipes/slda/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    sha256: aeabfe4edefc566ee97c107f8f9b2c6a478595e24fb35d44d9eeaaa876fb90ce
+
+build:
+    number: 0
+    skip: True  # [win or not py3k]
+
+requirements:
+    build:
+        - cython
+        - cythongsl
```
Comment (jakirkham) [inline recipes/slda/meta.yaml]: Does this mean we are dependent on `gsl`? Given `cythongsl` seems to be header only, could we please add `gsl` here too if we are linking to it?

File: recipes/slda/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        - numpy x.x
+        - python
+        - setuptools
+    run:
+        - cython
+        - cythongsl
```
Comment (jakirkham) [inline recipes/slda/meta.yaml]: Same as above.

File: recipes/slda/meta.yaml (hunk 6 above)
Comment (mmb90) [inline recipes/slda/meta.yaml]: > cythongsl depends on gsl. To avoid over-specifying the dependencies you can remove gsl from both build and run sections.

[quote](https://github.com/conda-forge/staged-recipes/pull/1018#discussion-diff-70455031) from @ocefpaf

**Comment by ocefpaf** [comment]
@mmb90 this PR slipped under the radar. (Sorry, too much noise in conda-forge org lately.)

It is good to go!

File: recipes/slda/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/slda/meta.yaml]: The problem with this is @mmb90 and @ocefpaf `slda` is actually [linking]( https://travis-ci.org/conda-forge/staged-recipes/jobs/146218765#L1129 ) to `gsl` unlike `cythongsl`. As a result, if `gsl` gets updated and we don't have a way to properly pin it here (e.g. `gsl`) is not listed, this will end up being broken. As this has been merged in spite of this comment, I would please ask you to fix it in the feedstock.

Comment (mmb90) [inline recipes/slda/meta.yaml]: Will do.

Comment (ocefpaf) [inline recipes/slda/meta.yaml]: The dependency tree is known we should improve our tools to do this instead of working around it by over-specifying.

Comment (jakirkham) [inline recipes/slda/meta.yaml]: Maybe that has merit in the general case. Happy to discuss that with you elsewhere. However I don't think it applies here.

`cythongsl` is a header only library and as such does not break when the ABI of `gsl` changes. However, `slda` that links to `gsl` will be broken if `gsl`'s ABI changes. I don't think we are over-specifying by providing this additional detail about ABI sensitivity.
